src/main/java/mezz/jei/ingredients/IngredientListElementFactory.java

```diff
@@ -23,7 +24,11 @@ private IngredientListElementFactory() {
 			addToBaseList(ingredientListElements, ingredientRegistry, ingredientType, modIdHelper);
 		}
 
-		ingredientListElements.sort(IngredientListElementComparator.INSTANCE);
+		try {
+			ingredientListElements.sort(IngredientListElementComparator.INSTANCE);
+		} catch (IllegalArgumentException ex) {
+			Log.get().error("Item sorting failed.  Aborting sort.", ex);
```

Comment:
Why catch `IllegalArgumentException` in particular? Are there other errors that can happen?
We need some kind of fallback behavior when this happens so that players still get some basic sorting.

Reply:
On a previous submission, I had been admonished for a generic catch.  This is the error thrown when timsort catches that a comparator violates the comparator contract.  I removed code that fell back to Quicksort.  Would you want that or force the sort to a basic default?  I favor a safe default.